apps/system/js/audio_channel_manager.js

```diff
@@ -0,0 +1,370 @@
+          this._handleAudioChannel({
+            isAllowedToPlay: false,
+            audioChannel: audioChannel
+          })
+          .then(this._resumeBackgroundAudioChannel.bind(this));
+        }
+      }
+      this._topMostWindow = Service.query('getTopMostWindow');
+      // Resume the foreground app's audio channels.
+      this.debug(this._topMostWindow.name + ' is opened');
+      this._resumeForegroundAudioChannels(this._topMostWindow);
+    },
+
+    /**
+     * Register the window's audio channels to be managed
+     * by AudioChannelManager when the window is created.
+     *
+     * @param {Event} evt The event to handle.
+     */
+    _handle_windowcreated: function(evt) {
+      var app = evt.detail;
+      app.registerAudioChannels();
+    },
+
+    /**
+     * Remove the window's audio channels from
+     * `_activeAudioChannels`, `_interruptedAudioChannels`,
+     * and `_audioChannelsResumeWhenOtherEnds`
+     * when the window is terminated.
+     *
+     * @param {Event} evt The event to handle.
+     */
+    _handle_windowterminated: function(evt) {
+      if (this._activeAudioChannels.size ||
+          this._interruptedAudioChannels.size) {
+        var app = evt.detail;
+        app.browser.element.allowedAudioChannels.forEach((audioChannel) => {
+          var id = app.getAudioChannel(audioChannel.name).instanceID;
+          this._activeAudioChannels.delete(id);
+          var element = this._interruptedAudioChannels.get(id);
+          if (element) {
+            this._interruptedAudioChannels.delete(id);
+            var index = element.backgroundResumeIndex;
+            index != null &&
+              this._audioChannelsResumeWhenOtherEnds.splice(index, 1);
+          }
+        });
+      }
+    },
+
+    /**
+     * Observer the value of vibration.enabled settings.
+     *
+     * @param {Boolean} value The value of the settings.
+     */
+    '_observe_vibration.enabled': function(value) {
+      this._isVibrateEnabled = value;
+    },
+
+    /**
+     * Play or pause the new audio channel and the active audio channels.
+     *
+     * @param {AudioChannelController} audioChannel The new audio channel.
+     * @return {Promise}
+     */
+    _manageAudioChannels: function(audioChannel) {
+      var promise;
+      var promises = [];
+      if (audioChannel.states.active) {
+        var policies = this.audioChannelPolicy.compete(
+          audioChannel,
+          this._activeAudioChannels,
+          {
+            isNewAudioChannelInBackground:
+              this._isAudioChannelInBackground(audioChannel)
+          }
+        );
+        var policiesForActive = policies.activeAudioChannels;
+        policiesForActive && policiesForActive.forEach((policy) => {
+          promises.push(this._handleAudioChannel(policy));
+        });
+        if (promises.length) {
+          promise = Promise.all(promises)
+            .then(this._handleAudioChannel(policies.newAudioChannel));
```

Comment:
policies.newAudioChannel -> audioChannel (you don't need AudioChannelPolicy to tell you who is the new one but you are just the one who tells it)